dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/LdapAdapter.java

```diff
@@ -207,6 +211,36 @@
      */
     private final AtomicReference<X509CertificateMapper> iCertificateMapperRef = new AtomicReference<X509CertificateMapper>();
 
+    /**
+     * Fetch the security service so we can load the KerberosService with any Kerberos config (keytab/config) for use if the bindAuthMechanism is GSSAPI/Kerberos
+     */
+    private final AtomicServiceReference<SecurityService> securityServiceRef = new AtomicServiceReference<SecurityService>("securityService");
+
+    /**
+     * Set the SecurityService ref
+     */
+    @Override
+    @Reference(name = KEY_SECURITY_SERVICE,
+               policy = ReferencePolicy.DYNAMIC,
+               cardinality = ReferenceCardinality.OPTIONAL)
+    protected void setSecurityService(ServiceReference<SecurityService> ref) {
+        securityServiceRef.setReference(ref);
+    }
+
+    /**
+     * Unset the SecurityService ref
+     */
+    @Override
+    protected void unsetSecurityService(ServiceReference<SecurityService> reference) {
+        securityServiceRef.unsetReference(reference);
+    }
+
+    /**
+     * The KerberosService is used, with any Kerberos config (keytab/config), if the bindAuthMechanism is GSSAPI/Kerberos
+     */
+    @Reference
+    protected KerberosService krb5Service;
```

Comment:
We should have a set / unset KerberosService method. Those methods should update the LdapConnection with the newly active KerberosService. Currently the activated method only sends in the active KerberosService on activation, and any changes afterwards are ignored. So if someone were to change the config file in the kerberos configuration, we would not pass that onto the LdapConnection.

Reply:
We need to support dynamic update of Kerberos configuration and Kerberos principal, ccache and keytab. If someone change the Kerberos configuration file, principal, ccache and keytab, we need to get a new connection for LDAP.

Reply:
Agree with Jesse if we don't need security service, then don't reference it.

Reply:
I will swap it back to just loading KerberosService and make sure modify works for reload (I think it did on my first round of testing, but will test it more thoroughly).